tests/unit/evaluation/SplittingStrategy_unittest.cc

```diff
@@ -297,89 +298,63 @@ TEST(SplittingStrategy,LOO)
 	}
 }
 
-TEST(SplittingStrategy, timeseries_subset_linear_splits)
+class TimeSeriesSplitTest : public ::testing::Test
 {
-	index_t num_labels, num_subsets, min_subset_size, base_size;
-	index_t runs = 10;
-
-	while (runs-- > 0)
+protected:
+	TimeSeriesSplitTest()
 	{
-		num_labels = CMath::random(50, 150);
-		num_subsets = CMath::random(1, 5);
-		min_subset_size = CMath::random(1, 6);
-		base_size = num_labels / num_subsets;
-
-		CRegressionLabels* labels = new CRegressionLabels(num_labels);
-		for (index_t i = 0; i < num_labels; ++i)
-			labels->set_label(i, CMath::random(-10.0, 10.0));
-
-		CTimeSeriesSplitting* splitting =
-		    new CTimeSeriesSplitting(labels, num_subsets);
+		num_labels = 15;
+		num_subsets = 3;
+		future_steps = 3;
 
-		splitting->set_min_subset_size(min_subset_size);
-		splitting->build_subsets();
-
-		for (index_t i = 0; i < num_subsets; ++i)
+		labels = new CRegressionLabels(num_labels);
```

Comment:
labels = some<CRegressionLabels>(num_labels); and then you dont have to SG_UNREF later